packages/roosterjs-editor-core/lib/editor/Editor.ts

```diff
@@ -436,10 +418,11 @@ export default class Editor {
     public insertContent(content: string, option?: InsertOption) {
         if (content) {
             let allNodes = fromHtml(content, this.core.document);
+
             // If it is to insert on new line, and there are more than one node in the collection, wrap all nodes with
             // a parent DIV before calling insertNode on each top level sub node. Otherwise, every sub node may get wrapped
             // separately to show up on its own line
-            if (option && option.insertOnNewLine && allNodes.length > 0) {
+            if (option && option.insertOnNewLine && allNodes.length > 1) {

```

Comment:
why was this number changed?

Reply:
See "Other fix 2" in the description:

Editor.insertContent() should not wrap a single node for "insertOnNewLine" scenario because it will call insertNode() which will wrap the node if it is not a block element.